cpp/tests/interpolate/cubic_spline_test.cpp

```diff
@@ -14,202 +14,105 @@
-            ids_column,
-            prefix_column
-        );
-
-    std::unique_ptr<cudf::column> interpolates = 
-        cuspatial::cubicspline_interpolate(
-            t_column,
-            point_ids_column,
-            prefix_column,
-            t_column,
-            splines->view()
-        );
-    
-    cudf::column_view device_column = interpolates->view();
-    std::vector<float> host_data;
-    host_data.resize(device_column.size());
-    cudaMemcpy(host_data.data(), device_column.data<float>(),
-          device_column.size() * sizeof(float),
-          cudaMemcpyDeviceToHost);
-
-    for(int i = 0 ; i < device_column.size() ; ++i){
-      EXPECT_EQ(x[i], host_data[i]);
-    }
-}
+TEST_F(CubicSplineTest, test_interpolate_single) {
+  cudf::test::fixed_width_column_wrapper<float> t_column{{0, 1, 2, 3, 4}};
+  cudf::test::fixed_width_column_wrapper<float> x_column{{3, 2, 3, 4, 3}};
+  cudf::test::fixed_width_column_wrapper<int> ids_column{{0, 0}};
+  cudf::test::fixed_width_column_wrapper<int> prefix_column{{0, 5}};
+  cudf::test::fixed_width_column_wrapper<int> point_ids_column{{0, 0, 0, 0, 0}};
 
-TEST_F(CubicSplineTest, test_interpolate_full)
-{
-    int point_len = 15;
-    float t[point_len] = {0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4};
-    float x[point_len] = {3, 2, 3, 4, 3, 3, 2, 3, 4, 3, 3, 2, 3, 4, 3};
-    int ids_len = 4;
-    int ids[ids_len] = {0, 0, 1, 2};
-    int prefix[ids_len] = {0, 5, 10, 15};
-    int point_ids[point_len] = {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2};
-
-    cudf::column t_column = make_device_column<float>(t, point_len);
-    cudf::column x_column = make_device_column<float>(x, point_len);
-    cudf::column ids_column = make_device_column<int>(ids, ids_len);
-    cudf::column prefix_column = make_device_column<int>(prefix, ids_len);
-    cudf::column point_ids_column = make_device_column<int>(point_ids, point_len);
-
-    std::unique_ptr<cudf::experimental::table> splines = 
-        cuspatial::cubicspline_coefficients(
-            t_column,
-            x_column,
-            ids_column,
-            prefix_column
-        );
-
-    std::unique_ptr<cudf::column> interpolates =
-        cuspatial::cubicspline_interpolate(
-            t_column,
-            point_ids_column.view(),
-            prefix_column,
-            t_column,
-            splines->view()
-        );
-    
-    cudf::column_view device_column = interpolates->view();
-    std::vector<float> host_data;
-    host_data.resize(device_column.size());
-    cudaMemcpy(host_data.data(), device_column.data<float>(),
-          device_column.size() * sizeof(float),
-          cudaMemcpyDeviceToHost);
-
-    for(int i = 0 ; i < device_column.size() ; ++i){
-      EXPECT_EQ(x[i], host_data[i]);
-    }
+  auto splines = cuspatial::cubicspline_coefficients(t_column, x_column,
+                                                     ids_column, prefix_column);
+
+  auto interpolates = cuspatial::cubicspline_interpolate(
```

Comment:
I think `interpolants` is the word.

```suggestion
  auto interpolants = cuspatial::cubicspline_interpolate(
```